### PR TITLE
*: preload load frequent use variable

### LINKED ISSUE #10388: avoid read kv for each new session's first `select @@variable`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

when user uses jdbc or spring, it will auto send many select variable statement like:

```
select @@tx_ready_only;
select @@net_buffer_length;
select @@max_allowed_packet;
select @@character_set_sever
....and so on
```

those statements will be frequent send, it's better preload their global variable value into TiDB memory.

We have some other variable already use this mechanism, but some other variable is missed(like above)

https://github.com/pingcap/tidb/blob/b0549b7875e9d9d7f14750a426a5775d6d4870da/session/session.go#L1614

3. What did you see instead?

frequent used variable's `select @@variable`, should read session variable or load default value from TiDB local cache, instead of load from kv everytime

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

3.0.1.beta1

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

fixes #10388, preload [JDBC](https://github.com/mysql/mysql-connector-j/blob/34cbc6bc61f72836e26327537a432d6db7c77de6/src/main/core-impl/java/com/mysql/cj/NativeSession.java) used variable into memory as default value 

this will improve to java based application that frequent establish new connections to TiDB.

### What is changed and how it works?

preload those variable:

```
	variable.InitConnect,
	variable.TxnIsolation,
	variable.TxReadOnly,
	variable.TransactionIsolation,
	variable.TransactionReadOnly,
	variable.NetBufferLength,
	variable.QueryCacheType,
	variable.QueryCacheSize,
	variable.CharacterSetServer,
	variable.AutoIncrementIncrement,
	variable.CollationServer,
	variable.NetWriteTimeout,
```

so, they can preload into session.

change `MaxAllowedPacket` as global + session scope  https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet

also change `InitConnect`, `QueryCacheSize` from global scope to global + session scope, this is not same with mysql, but doesn't break behave. why we need change this is TiDB will access kv every time if it's a global only variable, but for this two variable we no need do that in performance reason.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 
```
add log in GetGlobalSys method and start a jdbc app and connect to tidb and should not see any log output any more.
```

Code changes

 - impl change

Side effects

 - N/A

Related changes

 - N/A

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/10463)

### CODE REVIEW

File: executor/set_test.go [hunk 1]
```diff
@@ -325,6 +325,8 @@ func (s *testSuite2) TestSetVar(c *C) {
 	tk.MustQuery("select @@global.tx_isolation").Check(testkit.Rows("READ-UNCOMMITTED"))
 	tk.MustQuery("select @@global.transaction_isolation").Check(testkit.Rows("READ-UNCOMMITTED"))
 
+	tk.MustExec("SET GLOBAL transaction_isolation='REPEATABLE-READ'")
```
Comment (shenli) [inline executor/set_test.go]: What's this test case used for?

Comment (lysu) [reply executor/set_test.go]: we should reset tx_isolation back to rr before reset tidb_skip_isolation_level_check.

if we keep global tx_isolation in rc and reset tidb_skip_isolation_level_check to 0, when new session start and retry load global config into the session will meet `UnsupportedIsolationLevel` error.

Comment (lysu) [reply executor/set_test.go]: add some comments but maybe we should cascade reset tx_isolation to rr when tidb_skip_isolation_level_check be set to 0? :thinking:

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**